### DISCUSSION TITLE
Simple bluetooth status

### DISCUSSION THREAD

**Discussion by ToxicMushroom** [JakeStanger/ironbar#614] [Show and tell]
Hope this helps some people to serve as a starting point or inspiration.
Sadly I couldn't get battery level readings reliably

Sorry for the noob bash style/practices 

Connected
![image](https://github.com/JakeStanger/ironbar/assets/32853531/d57f48f4-29e5-4121-a609-3d1af3f5f932)

Powered on, not connected
![image](https://github.com/JakeStanger/ironbar/assets/32853531/4d9a3d0a-10c6-47c6-83dd-e993be2edf72)

Powered off
![image](https://github.com/JakeStanger/ironbar/assets/32853531/a8b6ee88-ed09-4361-b39d-e0c1dd6eb7c6)

No controller (like unplugging your adapter)
![image](https://github.com/JakeStanger/ironbar/assets/32853531/8575d68e-074a-42a8-a22c-475be05968ae)

```toml
[[end]]
type = "script"
cmd = "/home/merlijn/.bin/iron-bluetooth.sh"
mode = "poll"
interval = 5000
on_click_left = "blueman-manager"
```

The icons used require a nerdfont

```sh
BLU_CNT_CONTROLLER=$(ls -1q /sys/class/bluetooth | wc -l)

if [[ $BLU_CNT_CONTROLLER -gt 0 ]]
then
    BLU_POWER=$(bluetoothctl show | grep -q 'Powered: yes$'; echo "$?")
    if [[ $BLU_POWER -eq 0 ]]
    then
        BLU_CONNECTED=$(bluetoothctl devices Connected | sed -n -e 's/^Device \([[:xdigit:]]\{1,2\}:\)\{5\}[[:xdigit:]]\{1,2\} //p')
        BLU_CONNECTED_MAC=$(bluetoothctl devices Connected | sed -n -e 's/^Device \(\([[:xdigit:]]\{1,2\}:\)\{5\}[[:xdigit:]]\{1,2\}\) .*/\1/p')
        if [[ $BLU_CONNECTED != "" ]]
        then
            echo " $BLU_CONNECTED |"
        else
            echo " |"
        fi
    else 
        echo "󰂲 |"
    fi
else
    echo "󰂲 +  |"
fi
```

**Comment by bsag** [comment]
@ToxicMushroom Thanks so much for posting this - it really helped me to get started.

I managed to figure out the battery part via [Prime451's answer here](https://stackoverflow.com/questions/49078659/check-battery-level-of-connected-bluetooth-device-on-linux).

I have adapted your script to fit in with my use-case and paste it here in case it is useful to anyone else coming across this thread.

```shell
#!/usr/bin/env bash

BLU_CNT_CONTROLLER=$(ls /sys/class/bluetooth | wc -l)

if [[ $BLU_CNT_CONTROLLER -gt 0 ]]
then
    BLU_POWER=$(bluetoothctl show | grep -q 'Powered: yes$'; echo "$?")
    if [[ $BLU_POWER -eq 0 ]]
    then
        BLU_CONNECTED=$(bluetoothctl devices Connected | cut -d ' ' -f3-)
        BLU_CONNECTED_MAC=$(bluetoothctl devices Connected | cut -d ' ' -f2)
        DEV_BATTERY=$(echo "info $BLU_CONNECTED_MAC" | bluetoothctl | sed -n '/Battery Percentage:/ s/.*(\([0-9]*\).*/\1/p')
        if [[ $BLU_CONNECTED != "" ]]
        then
            echo " $BLU_CONNECTED $DEV_BATTERY%"
        else
            echo " None"
        fi
    else 
        echo "󰂲 Off"
    fi
else
    echo "󰂲 No controller"
fi
```

**Reply by ToxicMushroom** [reply]
Thanks this also works for me :)

**Comment by bsag** [comment]
![ironbar_middle_right](https://github.com/user-attachments/assets/e4df2959-5c68-459d-b57a-fc492ff6066e)

Here's a shot of the middle to right side so far.

**Reply by crimsonduelist** [reply]
u got dots for this?!

**Reply by bsag** [reply]
Not publicly right now, but I'll try to put some up when my installation has settled a bit.
